### ISSUE TITLE
Allow to configure line endings (and probably indentation)

### ISSUE THREAD

**Issue by digulla** [FasterXML/jackson-databind#585]
When writing multi-OS unit tests, it's cumbersome to filter the line endings from Jackson's output.

Please add a config option which allows to configure the line endings to use (and maybe the indentation level or String) so I can write unit tests which always produce a single Line Feed.

If I could configure the indentation, that would allow me to use tabs or four spaces to indent.

**Comment by derknorton** [comment]
Interesting discussion... I know this issue is already closed but it still sounds like a bug to me.  I use the `ObjectMapper` for serialization and deserialization of my POJOs into JSON as part of my jackson based RESTful web services. I don't know a way to do this with the `ObjectWriter`, perhaps its possible.

So I would like to add my own custom `PrettyPrinter` subclass that does not use the _inline_ formatting for arrays/collections while serializing the JSON for my web service. I need to do what @digulla was requesting above and have the `ObjectMapper` use my customer `PrettyPrinter` everywhere. It seems like this should be something that is possible and if not, perhaps should be considered a bug.

**Comment by cowtowncoder** [comment]
@derknorton what sounds like a bug? Pull request #166 was merged, included in 2.5.0.

As to semantics, while it's not a huge deal, missing functionality is not a bug, but, well, missing functionality. Creating a new issue for adding things that are missing and you would like to see is perfectly fine; and in fact I prefer it over re-opening old issues (unless it is specific flaw in fix/addition itself).

**Comment by james-bjss** [comment]
Realise this ticket is closed, but we faced a similar issue when matching messages in unit tests on Windows machines. So adding this here for others who stumble up on this page.

The `DefaultPrettyPrinter` was adding `\r\n` but we were expecting `\n`. 

The line-endings can be set as follows:

```
private static final ObjectWriter objectWriter = new ObjectMapper().writer(new DefaultPrettyPrinter()
            .withObjectIndenter(new DefaultIndenter().withLinefeed("\n")));
```
